ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -1564,17 +1564,20 @@ class JITModBuilder {
       // after this block finishes.
       size_t StackDepthBefore = CallStack.size();
 
+      // The index of the instruction in `BB` that we are currently processing.
+      size_t CurInstrIdx = 0;
+
+      // If we've returned from a call, skip ahead to the instruction where
+      // we left off.
+      std::optional<size_t> ResumeIdx = MPF->getResumeInstrIdx();
+      if (ResumeIdx.has_value()) {
+        CurInstrIdx = *ResumeIdx;
+        MPF->clearResume();
+      }
+
       // Iterate over all instructions within this block and copy them over
       // to our new module.
-      for (size_t CurInstrIdx = 0; CurInstrIdx < BB->size(); CurInstrIdx++) {
-        // If we've returned from a call, skip ahead to the instruction where
-        // we left off.
-        std::optional<size_t> ResumeIdx = MPF->getResumeInstrIdx();
-        if (ResumeIdx.has_value()) {
-          CurInstrIdx = *ResumeIdx;
-          MPF->clearResume();
-        }
-
+      for (; CurInstrIdx < BB->size(); CurInstrIdx++) {
```

Comment:
I'm generally not a fan of non-standard `for`s: I think this would be less likely to be misread as a `while`.

Reply:
If we make it a `while` then each `continue` will need an accompanying `CurInsrIdx++` which I think is even more risky than the non-standard while.

I count 7 `continue`s in the while loop.

Reply:
Fair enough!